### DISCUSSION TITLE
react-spring - react-spring/three - react-three/fiber conflict

### DISCUSSION THREAD

**Discussion by subaruMax** [pmndrs/react-spring#1625] [Support] [answered]
Hey. I am using react-spring - react-spring/three and react-three/fiber in my react SPA project.
The issues is: 
1. I have 3d scene created with react-three/fiber and react-spring/three on one of  my app pages. 
2. I wanted to use react-spring to animate some images etc. on another page.

The problem is that  i think react-spring and react-spring/three are conflicting each other, so that animation created with react-spring is not working at all, it's just frozen. 
If i remove react-three/fiber from 3D scene, animations created with react-spring will start to work, or if i create react-spring animations on the same page with 3D scene both of them working well.  
I was making research for a whole day, and can't find any answer, please help. 

I am using the following versions:
 "react-spring": "^9.2.1"
@react-spring/three": "^9.1.2"

Update: 
Just now discovered that if i load 3D scene with React.Lazy, everyting works fine, however this is not an option because 3D scene is my main intro to the page :(

**Comment by joshuaellis** [comment, marked as answer]
firstly, I would encourage you to use `@react-spring/web` for your web needs.
Secondly, `three` sets a Global that enforces the frameloop to be controlled by r3f. You can back of it of this to use it's own native `raf` by doing:

```js
Globals.assign({
  frameLoop: 'always',
})
```

I think this is an open issue, but no ones brought forward a good suggestion on how to resolve it so it's hanging there at the moment. Hopefully this helps you though!

**Reply by subaruMax** [reply]
Thanks alot, now everything works just fine!
